src/collector/rrdp.rs

```diff
@@ -1628,6 +1646,34 @@ impl TryFrom<IVec> for RepositoryObject<Bytes> {
 }
 
 
+//------------ FallbackTime --------------------------------------------------
+
+/// Parameters for calculating the best-before time of repositories.
+#[derive(Clone, Copy, Debug)]
+struct FallbackTime {
+    min: Duration,
+    max: Duration,
+}
+
+impl FallbackTime {
+    /// Creates a new value from the configuration.
+    pub fn from_config(config: &Config) -> Self {
+        FallbackTime {
+            min: config.refresh,
+            max: cmp::min(2 * config.refresh, config.rrdp_fallback_time)
+        }
+    }
+
+    /// Picks a best-before date for a repository updated around now.
+    pub fn best_before(self) -> DateTime<Utc> {
+        // Saturating conversion between std’s and chrono’s Duration types.
+        Utc::now() + chrono::Duration::from_std(
+            rand::thread_rng().gen_range(self.min..self.max)
+        ).unwrap_or_else(|_| chrono::Duration::milliseconds(i64::MAX))
```

Comment:
Does this `_or_else` mean that in case anything goes wrong, the best_before date will be very far in the future and the fallback to rsync will realistically never happen? Why not default back to half an hour or an hour worth of milliseconds, in that error case?

Reply:
The conversation can only go wrong if the std duration is too big. So this just cuts it back to whatever the biggest value is. Not perfect but also very theoretical.